frappe/model/document.py

```diff
@@ -836,6 +836,8 @@ def _submit(self):
 		"""Submit the document. Sets `docstatus` = 1, then saves."""
 		self.docstatus = 1
 		self.save()
+		self.db_set("submission_date", now())
```

Comment:
@rkurungadam why not set this before `save()` the same way as `docstatus`

Reply:
ensure that it accepts `submission_date` and `submitted_by` as legit.

Reply:
had to add the fields explicitly in meta,  not sure if its the right way.